pycytominer/cyto_utils/cells.py

```diff
@@ -430,46 +435,210 @@ def aggregate_compartment(
+                subset_data_df=self.subset_data_df,
+                features=aggregate_features,
+                object_feature=self.object_feature,
             )
-        else:
-            aggregate_features = self.features
-
-        object_df = aggregate(
-            population_df=population_df,
-            strata=self.strata,
-            compute_object_count=compute_counts,
-            operation=self.aggregation_operation,
-            subset_data_df=self.subset_data_df,
-            features=aggregate_features,
-            object_feature=self.object_feature,
-        )
 
-        if compute_counts and self.fields_of_view_feature not in self.strata:
-            fields_count_df = self.image_df.loc[
-                :, list(np.union1d(self.strata, self.fields_of_view_feature))
-            ]
-            fields_count_df = (
-                fields_count_df.groupby(self.strata)[self.fields_of_view_feature]
-                .count()
-                .reset_index()
-                .rename(
-                    columns={f"{self.fields_of_view_feature}": f"Metadata_Site_Count"}
+            if compute_counts and self.fields_of_view_feature not in self.strata:
+                fields_count_df = self.image_df.loc[
+                    :, list(np.union1d(self.strata, self.fields_of_view_feature))
+                ]
+                fields_count_df = (
+                    fields_count_df.groupby(self.strata)[self.fields_of_view_feature]
+                    .count()
+                    .reset_index()
+                    .rename(
+                        columns={
+                            f"{self.fields_of_view_feature}": f"Metadata_Site_Count"
+                        }
+                    )
                 )
-            )
 
-            object_df = fields_count_df.merge(object_df, on=self.strata, how="right")
+                partial_object_df = fields_count_df.merge(
+                    partial_object_df,
+                    on=self.strata,
+                    how="right",
+                )
+
+            object_dfs.append(partial_object_df)
+
+        # Concatenate one or more aggregated dataframes row-wise into final output
+        object_df = pd.concat(object_dfs, axis=0).reset_index(drop=True)
 
         return object_df
 
+    def _compartment_df_generator(
+        self,
+        compartment,
+        n_strata=1,
+    ):
+        """A generator function that returns chunks of the entire compartment
+        table from disk.
+
+        We want to return dataframes with all compartment entries within unique
+        combinations of self.merge_cols when aggregated by self.strata
+
+        Parameters
+        ----------
+        compartment : str
+            Compartment to aggregate.
+        n_strata : int, default 1
+            Number of unique strata to pull from the database into working memory
+            at once.  A larger number does not lead to faster compute times
+
+        Yields
```

Comment:
This might actually not work as a sphinx autodoc doctoring title, I didn't check actually.  It looks like the right thing to do might be to use "Returns" with an `Iterator` specified, as in
https://stackoverflow.com/questions/24461529/is-there-a-sphinx-rest-python-docstring-field-for-yields

Reply:
"Yields" might work, I'm not sure
https://www.sphinx-doc.org/en/master/usage/extensions/example_google.html
I'm not super experienced at this part...